presto-spi/src/main/java/io/prestosql/spi/block/LazyBlock.java

```diff
@@ -276,6 +283,10 @@ private void assureLoaded()
         }
         loader.load(this);
```

Comment:
```suggestion
        loader.load(this).getLoadedBlock();
```
However I think you should just assert here that loaded block is truly loaded as `LazyBlockLoader#load` implicitly loads fully (see implementations)

Reply:
`assureLoaded` has always only loaded a single level and is not fully recursive. The only change here is to load through single chains of lazy blocks, so that the new `getBlock()` method is assured not to return a lazy block (but can return a container block that contains lazy blocks).